### PR TITLE
Bootstrapping disco-gapic

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Baseline tests are not bootstrapped yet. 

Note - this is a PR against the discogapic branch, and is meant to unblock @andreamlin as opposed to being fully clean and fully commented.

### CODE REVIEW

**Comment by vchudnov-g** [comment]
Apologies for the delay; I've been distracted by other things. I plan to spend time on this tomorrow.

File: src/main/java/com/google/api/codegen/discogapic/DiscoGapicProvider.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import com.google.api.codegen.discogapic.transformer.DocumentToViewTransformer;
+import com.google.api.codegen.discovery.Document;
+import com.google.api.codegen.rendering.CommonSnippetSetRunner;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.tools.framework.snippet.Doc;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class DiscoGapicProvider {
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/discogapic/DiscoGapicProvider.java]: class comment

File: src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import com.google.api.codegen.gapic.CommonGapicCodePathMapper;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.gapic.GapicGeneratorConfig;
+import com.google.api.codegen.rendering.CommonSnippetSetRunner;
+import com.google.api.codegen.util.java.JavaRenderingUtil;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.NotImplementedException;
+
+public class MainDiscoGapicProviderFactory implements DiscoGapicProviderFactory {
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java]: "Main" does not seem very informative. What other implementations of `DiscoGapicProviderFactory` are envisioned?

File: src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                .setDocumentToViewTransformer(
+                    new JavaDiscoGapicSurfaceTransformer(javaPathMapper, packageConfig))
+                .build();
+
+        providers.add(mainProvider);
+      }
+      return providers;
+
+    } else {
+      throw new NotImplementedException("DiscoGapicProviderFactory: invalid id \"" + id + "\"");
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java]: To be more informative, this message should reflect the class name, not the interface name.

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSurfaceTransformer.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import java.util.List;
+
+public class JavaDiscoGapicSurfaceTransformer implements DocumentToViewTransformer {
+  private final GapicCodePathMapper pathMapper;
+  private final StandardImportSectionTransformer importSectionTransformer =
+      new StandardImportSectionTransformer();
+  private final FileHeaderTransformer fileHeaderTransformer =
+      new FileHeaderTransformer(importSectionTransformer);
+
+  private static final String XAPI_TEMPLATE_FILENAME = "java/main.snip";
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSurfaceTransformer.java]: Why the `X` prefixes?

**Comment by garrettjonesgoogle** [comment]
@andreamlin will take care of the comments in a following PR. Merging now.

**Pull request merged**